File: c_program00.0_soloc/testMatrix/matrixToFile.f90
program test
IMPLICIT NONE

INTEGER i,j,k
REAL, DIMENSION( 1:5 , 1:5 , 1:5 ) :: matrix
REAL counter

counter = 0

DO k = 1, 5
DO j = 1, 5
DO i = 1, 5
	!matrix(i,j,k) = counter
	!counter = counter + 1
	matrix(i,j,k) = j
ENDDO
ENDDO
ENDDO


print *, matrix(1:5,1:5,1:5)    

OPEN(UNIT=10, FILE="outf.txt", ACTION="write", STATUS="replace")
WRITE(10,*) matrix
CLOSE(UNIT=10)
 

stop
end program test

File: c_program00.0_soloc/testMatrix/matrixToFile.c
#include<stdio.h>

int main()
{
	int i,j,k;
	float matrix[125];
	int counter = 0;

	for(k=0; k<5; k++){
		for(j=0; j<5; j++){
			for(i=0; i<5; i++){
				//matrix[i+j*5+k*25] = counter;
				matrix[i+j*5+k*25] = j;
				counter++;
			}
		}
	}

	for(k=0; k<5; k++){
		for(j=0; j<5; j++){
			for(i=0; i<5; i++){
				printf("%f\t", matrix[i+j*5+k*25]);
			}
		}
	}
	
	FILE *f = fopen("outc.txt", "wb");
	fwrite(matrix, sizeof(float), sizeof(matrix), f);
	fclose(f);

	return 0;
}
